src/Microsoft.VisualStudio.ProjectSystem.IntegrationTests/OpenProjectTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.VisualStudio.ProjectSystem.IntegrationTests
+{
+    [TestClass]
+    public class CreateProjectTests : VisualStudioHostTest
+    {
+        private static string _hiveName;
+
+        [ClassInitialize()]
+        public static void ClassInitialize(TestContext context)
+        {
+            _hiveName = context.Properties["VsRootSuffix"].ToString();
+        }
+
+        protected VisualStudioHostConfiguration DefaultHostConfiguration
+        {
+            get
+            {
+                var visualStudioHostConfiguration = new VisualStudioHostConfiguration()
+                {
+                    CommandLineArguments = $"/rootSuffix:{_hiveName}",
+                    RestoreUserSettings = false,
+                    InheritProcessEnvironment = true,
+                    AutomaticallyDismissMessageBoxes = true,
+                    DelayInitialVsLicenseValidation = true,
+                    ForceFirstLaunch = true,
+                };
+
+                return visualStudioHostConfiguration;
+            }
+        }
+
+        protected VisualStudioHost GetVS() => Operations.CreateHost<VisualStudioHost>(DefaultHostConfiguration);
+
+        [TestMethod]
+        public void CreateProject_CreateRestoreAndBuild()
+        {
+            var VS = GetVS();
+
+            VS.Start();
+
+            using (Scope.Enter("Create Project"))
+            {
+                var proj = VS.ObjectModel.Solution.CreateProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreConsoleApp);
+            }
+
+            using (Scope.Enter("Verify Create Project"))
+            {
+                VS.ObjectModel.Solution.Verify.HasProject();
+            }
+
+            using (Scope.Enter("Wait For Restore"))
+            {
+                VS.ObjectModel.Shell.ToolWindows.ErrorList.WaitForReady();
+                VS.ObjectModel.Shell.ToolWindows.ErrorList.WaitForNoErrorListItems();
```

Comment:
Does this actually work? I would expect not to work due to the Design-time build -> restore -> design-time build cycle.

Reply:
It works in practice but not in theory.  Until we have a real way to track restore operations I think I will remove this.